lib/layouts.js

```diff
@@ -210,8 +212,10 @@ function patternLayout(pattern, tokens) {
     return os.hostname().toString();
   }
 
-  function formatMessage(loggingEvent) {
-    return util.format(...loggingEvent.data);
+  function formatMessage(loggingEvent, specifier) {
+    const [lowerBound, upperBound] = specifier ? specifier.split(',') : [];
+    const dataSlice = loggingEvent.data.slice(lowerBound, upperBound);
+    return util.format(...dataSlice);
```

Comment:
To save calling `dataSlice.slice(undefined, undefined)` when specifier is `undefined` which defaults to `[]`.

```suggestion
    let dataSlice = loggingEvent.data;
    if (specifier) {
      const [lowerBound, upperBound] = specifier.split(',');
      dataSlice = dataSlice.slice(lowerBound, upperBound);
    }
    return util.format(...dataSlice);
```
Anyway, this change will also be consistent with the existing `function categoryName()`.
https://github.com/log4js-node/log4js-node/blob/43bdea724337d7cf98f2953335792aeac6e670e4/lib/layouts.js#L142-L154